src/Blocks/components/form/assets/form.js

```diff
@@ -483,43 +483,43 @@ export class Form {
 		}
 
 		// Add form ID field.
-		formData.append('es-form-post-id', JSON.stringify({
-			value: element.getAttribute(this.DATA_ATTR_FORM_POST_ID),
+		formData.append(this.FORM_CUSTOM_FORM_PARAMS.postId, JSON.stringify({
+			value: element.getAttribute(FORM_DATA_ATTRIBUTES.DATA_ATTR_FORM_POST_ID),
 			type: 'hidden',
 		}));
 
 		// Add form type field.
-		formData.append('es-form-type', JSON.stringify({
+		formData.append(this.FORM_CUSTOM_FORM_PARAMS.type, JSON.stringify({
 			value: formType,
 			type: 'hidden',
 		}));
 
 		// Add form action field.
-		formData.append('action', JSON.stringify({
-			value: formAction,
+		formData.append(this.FORM_CUSTOM_FORM_PARAMS.action, JSON.stringify({
+			value: element.getAttribute('action'),
 			type: 'hidden',
 		}));
 
 		// Add additional options for HubSpot only.
 		if (formType === 'hubspot' && !this.formIsAdmin) {
-			formData.append('es-form-hubspot-cookie', JSON.stringify({
+			formData.append(this.FORM_CUSTOM_FORM_PARAMS.hubspotCookie, JSON.stringify({
```

Comment:
Might be a stupid question but why not separate integration specifics to each particular file, for example`form-hubspot.js`. Also other functionalities like GTM events and similar. 
That way we would prevent 1000 LOC of initial load and split it into asynchronous fetching, and ensure that all that you are changing is the functionality that is not connected to any other part of the code. If you have to rework the event triggering in GTM, you work on GTM events etc.



Potentially, that way this file could look like:

```js
setEventListeners(...);
prepFields(...)
beforeFormSubmit(...);
formSubmit(...);
afterFormSubmit(...);
```
and with different params and data you decide what is going on in each of the integrations.

Reply:
there are just a few small stuffs that are implementation specific so I don't see the need for this at this point. I will reconsider this when we do conditional tags and multi step form. tnx for the idea.